essentials/custom_node_more_on_inputs.mdx

```diff
@@ -0,0 +1,101 @@
+---
+title: "Hidden and Flexible inputs"
+---
+
+## Hidden inputs
+
+Alongside the `required` and `optional` inputs, which create corresponding inputs or widgets on the client-side,
+there are three `hidden` input options which allow the custom node to request certain information from the server.
+
+These are accessed by returning a value for `hidden` in the `INPUT_TYPES` `dict`, with the signature `dict[str,str]`,
+containing one or more of `PROMPT`, `EXTRA_PNGINFO`, or `UNIQUE_ID`
+
+```python
+@classmethod
+def INPUT_TYPES(s):
+    return {
+        "required": {...},
+        "optional": {...},
+        "hidden": {
+            "prompt": "PROMPT", 
+            "extra_pnginfo": "EXTRA_PNGINFO",
+            "unique_id": "UNIQUE_ID",
+        }
+    }
+```
+
+### UNIQUE_ID 
+`UNIQUE_ID` is the unique identifier of the node, and matches the `id` property of the node on the client side. 
+It is commonly used in client-server communications (see [messages](./comms_messages#getting-node-id)).
+
+### PROMPT
+`PROMPT` is the complete prompt sent by the client to the server. 
+See [the prompt object](./javascript_objects_and_hijacking#prompt) for a full description.
+
+### EXTRA_PNGINFO
+`EXTRA_PNGINFO` is a dictionary that will be copied into the metadata of any `.png` files saved. Custom nodes can store additional
+information in this dictionary for saving (or as a way to communicate with a downstream node).
+
+<Tip>Note that if Comfy is started with the `disable_metadata` option, this data won't be saved.</Tip>
```

Comment:
```suggestion
<Tip>Note that if Comfy is started with the `--disable_metadata` option, this data won't be saved.</Tip>
```

Reply:
Also a good link target if there's a page with all the options listed.

Reply:
Added this as a TODO comment for later searching!